pilot/pkg/bootstrap/configcontroller.go

```diff
@@ -366,20 +366,49 @@ func (s *Server) makeFileMonitor(fileDir string, domainSuffix string, configCont
 // Implemented only for SIMPLE_TLS mode
 // TODO:
 //
-//	Implement for MUTUAL_TLS/ISTIO_MUTUAL_TLS modes
+//	Implement for ISTIO_MUTUAL_TLS mode
 func (s *Server) getTransportCredentials(args *PilotArgs, tlsSettings *v1alpha3.ClientTLSSettings) (credentials.TransportCredentials, error) {
-	if err := agent.ValidateTLS(tlsSettings); err != nil && tlsSettings.GetMode() == v1alpha3.ClientTLSSettings_SIMPLE {
+	if err := agent.ValidateTLS(tlsSettings); err != nil {
 		return nil, err
 	}
 	switch tlsSettings.GetMode() {
 	case v1alpha3.ClientTLSSettings_SIMPLE:
 		if len(tlsSettings.GetCredentialName()) > 0 {
-			rootCert, err := s.getRootCertFromSecret(tlsSettings.GetCredentialName(), args.Namespace)
+			rootCert, _, err := s.getCertInfoFromSecret(tlsSettings.GetCredentialName(), args.Namespace)
+			if rootCert == nil {
+				return nil, err
+			}
+			tlsSettings.CaCertificates = string(rootCert.Cert)
+			tlsSettings.CaCrl = string(rootCert.CRL)
+		}
+		if tlsSettings.GetInsecureSkipVerify().GetValue() || len(tlsSettings.GetCaCertificates()) == 0 {
+			return credentials.NewTLS(&tls.Config{
+				ServerName:         tlsSettings.GetSni(),
+				InsecureSkipVerify: tlsSettings.GetInsecureSkipVerify().GetValue(), //nolint
```

Comment:
Missed this before. I think no CaCertificates set should be to use the system certs, not insecure.